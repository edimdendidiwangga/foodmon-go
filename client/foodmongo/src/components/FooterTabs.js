import React, { Component } from 'react';
import { Footer, FooterTab, Button, Icon, Text } from 'native-base';
import { Actions } from 'react-native-router-flux';


class FooterTabs extends Component {

  onPressHandler = (key) => {
    console.log(`onPress ${key}`)
    // Actions.map({type: "reset"})
    if (key === 'map') {
      console.log(Actions);
      Actions.pop();
    } else if (key === 'signup') {
      Actions.signup();
    } else if (key === 'signin') {
      Actions.signin();
    } else {
      Actions.map();
    }
  }

  render() {
    return (
      <Footer >
        <FooterTab>
          <Button active vertical onPress={() => { this.onPressHandler('map'); }}>
            <Icon active name="navigate" />
            <Text>Explore</Text>
          </Button>
          <Button vertical>
            <Icon name="apps" />
            <Text>Apps</Text>
          </Button>
          <Button vertical>
            <Icon name="camera" />
            <Text>Camera</Text>
          </Button>
          <Button vertical onPress={() => { this.onPressHandler('signin'); }}>
            <Icon name="person" />
            <Text>Sign Up</Text>
          </Button>
        </FooterTab>
      </Footer>
    );
  }
}

export default FooterTabs;
