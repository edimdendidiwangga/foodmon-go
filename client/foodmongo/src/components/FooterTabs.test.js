import renderer from 'react-test-renderer';
import React from 'react';

import { FooterTabs } from './FooterTabs'

describe('FooterTabs', () => {

  it('renders and match to snapshot', () => {
    const tree = renderer.create(<FooterTabs user_login={{ username: '' }} nav_counter={0} addCounter={() => {}} subtractCounter={() => {}} resetCounter={() => {}} />);
    expect(tree).toMatchSnapshot();
  });
});
