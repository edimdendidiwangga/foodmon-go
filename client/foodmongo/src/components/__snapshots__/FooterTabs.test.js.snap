// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterTabs renders and match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F8F8",
      "borderColor": "#cbcbcb",
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexDirection": "row",
      "height": 55,
      "justifyContent": "center",
      "left": 0,
      "right": 0,
    }
  }
  virtual={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": undefined,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#cde1f9",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#007aff",
              "fontSize": 24,
              "marginLeft": 5,
              "marginRight": 5,
              "paddingTop": 2,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#007aff",
            "fontFamily": "System",
            "fontSize": 14,
            "lineHeight": 16,
            "marginLeft": 0,
            "marginRight": 0,
          }
        }
        virtual={undefined}
      >
        Explore
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#6b6b6b",
              "fontSize": 24,
              "marginLeft": 5,
              "marginRight": 5,
              "paddingTop": 2,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#6b6b6b",
            "fontFamily": "System",
            "fontSize": 14,
            "lineHeight": 16,
            "marginLeft": 0,
            "marginRight": 0,
          }
        }
        virtual={undefined}
      >
        Interests
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#6b6b6b",
              "fontSize": 24,
              "marginLeft": 5,
              "marginRight": 5,
              "paddingTop": 2,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#6b6b6b",
            "fontFamily": "System",
            "fontSize": 14,
            "lineHeight": 16,
            "marginLeft": 0,
            "marginRight": 0,
          }
        }
        virtual={undefined}
      >
        List
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#6b6b6b",
              "fontSize": 24,
              "marginLeft": 5,
              "marginRight": 5,
              "paddingTop": 2,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#6b6b6b",
            "fontFamily": "System",
            "fontSize": 14,
            "lineHeight": 16,
            "marginLeft": 0,
            "marginRight": 0,
          }
        }
        virtual={undefined}
      >
        Sign In
      </Text>
    </View>
  </View>
</View>
`;
